aten/src/ATen/native/mobile/cpu/Engine.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+// Prefer not to use these functions directly.  Instead prefer to use the
```

Comment:
I mean, you're not supposed to call functions in the `native::` namespace anyway :)

Reply:
Typically, most functions in `native` live in `cpp` headers without corresponding headers (which are generated directly by the codegen). I'd like to know why this isn't the case here.

Reply:
Originally I was thinking of providing two code paths to reach these functions: one through _convolution and the other through c10.  These declarations would enable the former.  Registration with c10 were done inside the .cpp files, which I have now removed in favor of adding them through native_functions.  My hope was that XNNPACK could replace NNPACK to simplify the code.  It is faster for NHWC always, compared to the alternative of converting NHWC to NCHW and going through NNPACK, even without caching (meaning calling weight prepacking at forward() time repeatedly which is wasteful), and sometimes faster than NCHW even after the extra NHWC to NCHW conversion to allow usage through XNNPACK.  So the plan was to remove NNPACK, and expose this codepath via _convolution for the less efficient usage (still always better NHWC, and sometimes better for NCHW as mentioned previously), and expose the create and run operators separately directly (think conv_prepack and conv) for optimal performance.

Reply:
> so the plan was to remove NNPACK, and expose this codepath via _convolution for the less efficient usage (still always always faster for NHWC, and sometimes faster for NCHW as mentioned previously)

So, by a quirk of how `_convolution` is setup, you don't need the header declarations for access via `_convolution` either. This is because every subsidiary implementation `_convolution` calls (e.g., `cudnn_convolution`) is also defined in `native_functions.yaml` and gets its headers generated for you. So it's better not to define the headers for it :)

Reply:
Sure, I'll remove the header.